app/Helpers/database/user-auth.php

```diff
@@ -180,6 +180,7 @@ function authenticateFromCookie(
     }
 
     $userDetailsOut = $user->toArray();
+    $userDetailsOut['isMuted'] = $user->isMuted;
```

Comment:
Why is this necessary? shouldn't it already be populated by the `toArray` call since it's in the `visible` list?

Reply:
The necessity comes down to how Laravel handles attribute serialization for accessors. Although `isMuted` is indeed set up as an accessor in the model and is included in the `$visible` list, because the value is driven by an accessor and is dynamically computed, it also needs to be added to `$appends` for `->toArray()` to serialize it, eg:

```php
protected $appends = [
  'avatarUrl',
  'isMuted',
];
```

However, I'm afraid this can result in the following unexpected developer experience:

```php
$user = User::select("User", "Motto")->firstWhere("User", "Scott");

// ERROR: The attribute [muted_until] either does not exist or 
// was not retrieved for model [App\Models\User].
$user->toArray();
```

I think maybe we should just remove it from `$visible`.